File: Modules/module_jetpack/src/main/java/com/example/william/my/module/jetpack/comparator/ArticleComparator.java
package com.example.william.my.module.jetpack.comparator;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import com.example.william.my.module.bean.ArticleBean;

public class ArticleComparator extends DiffUtil.ItemCallback<ArticleBean.DataBean.ArticleDetailBean> {

    @Override
    public boolean areItemsTheSame(@NonNull ArticleBean.DataBean.ArticleDetailBean oldItem, @NonNull ArticleBean.DataBean.ArticleDetailBean newItem) {
        return oldItem.getId() == newItem.getId();
    }

    @Override
    public boolean areContentsTheSame(@NonNull ArticleBean.DataBean.ArticleDetailBean oldItem, @NonNull ArticleBean.DataBean.ArticleDetailBean newItem) {
        return oldItem.getId() == newItem.getId();
    }
}

File: modules/module_kotlin/src/main/kotlin/com/example/william/my/module/kotlin/comparator/ArticleComparator.kt
package com.example.william.my.module.kotlin.comparator

import androidx.recyclerview.widget.DiffUtil
import com.example.william.my.bean.data.ArticleDetailBean

class ArticleComparator :
    DiffUtil.ItemCallback<ArticleDetailBean>() {

    override fun areItemsTheSame(
        oldItem: ArticleDetailBean,
        newItem: ArticleDetailBean
    ): Boolean {
        // Id is unique.
        return oldItem.id == newItem.id
    }

    override fun areContentsTheSame(
        oldItem: ArticleDetailBean,
        newItem: ArticleDetailBean
    ): Boolean {
        return newItem.id == oldItem.id
    }
}